### PR TITLE
Block stewards read access to staffwiki

### PR DESCRIPTION
**Pull request by paladox**

### CODE REVIEW

File: includes/MirahezeMagicHooks.php [hunk 1]
```diff
@@ -254,4 +254,23 @@ public static function onSkinAddFooterLinks( Skin $skin, string $key, array &$fo
...
+
+	public static function onUserGetRightsRemove( User $user, array &$aRights ) {
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'mirahezemagic' );
+		// Remove read from stewards on staff wiki.
+		if ( $config->get( 'DBname' ) === 'staffwiki' && $user->isLoggedIn() &&
+		    class_exists( 'CentralAuthUser' ) ) {
+			$centralAuthUser = CentralAuthUser::getInstance( $user );
+			if ( $centralUser->exists() && $centralUser->isAttached() &&
+			    isset( $centralAuthUser->getGlobalGroups()['steward'] ) ) {
+				foreach ( $aRights as $i => $right ) {
```
Comment (Universal-Omega) [inline includes/MirahezeMagicHooks.php:266]: Why not just $aRights['read'] ?? unset or similar?

File: includes/MirahezeMagicHooks.php [hunk 2]
```diff
@@ -254,4 +254,23 @@ public static function onSkinAddFooterLinks( Skin $skin, string $key, array &$fo
...
+					if ( $right == 'read' ) {
+						unset( $rights[$i] );
+					}
+				}
+			}
+		}
+
+		return true;
```
Comment (Universal-Omega) [inline includes/MirahezeMagicHooks.php:274]: Not necessary

File: includes/MirahezeMagicHooks.php [hunk 3]
```diff
@@ -254,4 +254,23 @@ public static function onSkinAddFooterLinks( Skin $skin, string $key, array &$fo
 			$footerItems['donate'] = $skin->footerLink( 'miraheze-donate', 'miraheze-donatepage' );
 		}
 	}
+
+	public static function onUserGetRightsRemove( User $user, array &$aRights ) {
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'mirahezemagic' );
+		// Remove read from stewards on staff wiki.
+		if ( $config->get( 'DBname' ) === 'staffwiki' && $user->isLoggedIn() &&
+		    class_exists( 'CentralAuthUser' ) ) {
```
Comment (Universal-Omega) [inline includes/MirahezeMagicHooks.php:262]: Why not ExtensionRegistry->isLoaded ?

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

File: includes/MirahezeMagicHooks.php (hunk 2 above)
Comment (paladox) [reply includes/MirahezeMagicHooks.php]: Thanks, done.

File: includes/MirahezeMagicHooks.php (hunk 3 above)
Comment (paladox) [reply includes/MirahezeMagicHooks.php]: Decided to remove the check as we check for the wiki we want this done on.

File: includes/MirahezeMagicHooks.php (hunk 1 above)
Comment (paladox) [reply includes/MirahezeMagicHooks.php]: Changed the logic thus that is unneeded.

**Review by Universal-Omega — APPROVED** [review]

File: includes/MirahezeMagicHooks.php [hunk 4]
```diff
@@ -254,4 +254,19 @@ public static function onSkinAddFooterLinks( Skin $skin, string $key, array &$fo
...
 		}
 	}
+
+	public static function onUserGetRightsRemove( User $user, array &$aRights ) {
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'mirahezemagic' );
+		// Remove read from stewards on staff wiki.
+		if ( $config->get( 'DBname' ) === 'staffwiki' && $user->isLoggedIn() ) {
+			$centralAuthUser = CentralAuthUser::getInstance( $user );
+			if ( $centralAuthUser &&
+			    $centralUser->exists() &&
```
Comment (The-Voidwalker) [inline includes/MirahezeMagicHooks.php:264]:
```suggestion
			    $centralAuthUser->exists() &&
```

**Review by The-Voidwalker — CHANGES_REQUESTED** [review]

**Pull request merged**